featran-jmh/src/test/scala/com/spotify/featran/jmh/HotEncoderTransformersBenchmark.scala

```diff
@@ -0,0 +1,193 @@
+  //
+  @Benchmark
+  def nHotRequiredWithListMatch: Unit =
+    nHotRequiredWithListMatchPrepare(arbitraryListOptional)
+
+  @Benchmark
+  def nHotRequiredWithListMatch10Values: Unit =
+    nHotRequiredWithListMatchPrepare(arbitraryListOptional10Values)
+
+  @Benchmark
+  def nHotRequiredWithListMatch1000Values: Unit =
+    nHotRequiredWithListMatchPrepare(arbitraryListOptional1000Values)
+
+  //
+  // ONE HOT - required
+  //
+  @Benchmark
+  def oneHotRequiredEncoder: Unit = oneHotRequired(arbitraryString)
+
+  @Benchmark
+  def oneHotRequiredEncoder1000Values: Unit =
+    oneHotRequired(arbitraryString1000Values)
+
+  //
+  // ONE HOT - optional
+  //
+  @Benchmark
+  def oneHotOptionalEncoder: Unit = oneHotOptional(arbitraryString)
+
+  @Benchmark
+  def oneHotOptionalEncoder10Values: Unit =
+    oneHotOptional(arbitraryString10Values)
+
+  @Benchmark
+  def oneHotOptionalEncoder10SameValues: Unit =
+    oneHotOptional(arbitraryString10SameValues)
+
+  @Benchmark
+  def oneHotOptionalEncoder1000Values: Unit =
+    oneHotOptional(arbitraryString1000Values)
+
+  @Benchmark
+  def oneHotOptionalEncoder1000SameValues: Unit =
+    oneHotOptional(arbitraryString1000SameValues)
+
+  def nHotOptional(
+      input: List[List[String]]): FeatureExtractor[List, List[String]] = {
+    val featureSpec = FeatureSpec
+      .of[List[String]]
+      .optional {
+        case List("") => None
+        case i        => Option(i)
+      }(NHotEncoder("n_hot"))
+    featureSpec.extract(input)
+  }
+
+  def nHotRequired(
+      input: List[List[String]]): FeatureExtractor[List, List[String]] = {
+    val featureSpec = FeatureSpec
+      .of[List[String]]
+      .required(identity)(NHotEncoder("n_hot"))
+    featureSpec.extract(input)
+  }
+
+  def nHotRequiredWithListMatchPrepare(
+      input: List[Option[String]]): FeatureExtractor[List, List[String]] = {
+    val listInput = input.map {
+      case Some("") => List()
+      case Some(i)  => List(i)
+      case _        => List()
+    }
+    val featureSpec = FeatureSpec
+      .of[List[String]]
+      .required(identity)(NHotEncoder("n_hot"))
+    featureSpec.extract(listInput)
+  }
+
+  def oneHotOptional(input: List[String]): FeatureExtractor[List, String] = {
+    val featureSpec = FeatureSpec
+      .of[String]
+      .optional {
+        case "" => None
+        case f  => Option(f)
+      }(OneHotEncoder("one_hot"))
+    featureSpec.extract(input)
+  }
+
+  def oneHotRequired(input: List[String]): FeatureExtractor[List, String] = {
+    val featureSpec = FeatureSpec
+      .of[String]
+      .required(identity)(OneHotEncoder("one_hot"))
+    featureSpec.extract(input)
+  }
+
+  private val thisString = Gen.alphaStr.toString
+
+  // Lists of Lists for N-Hot
+  private implicit val arbitraryListString: List[List[String]] = List(
+    List(Gen.alphaStr.toString))
+  private implicit val arbitraryListString10Values: List[List[String]] = List(
+    Range(0, 10).map(_ => Gen.alphaStr.toString).toList)
```

Comment:
Not sure `Gen[T]#toString` does what you want. `def sample: Option[T]` is probably what you want (generates another random sample).
I'd just ditch scalacheck and hardcode all input to make it more predictable.